server/index.js

```diff
@@ -90,9 +90,70 @@ app.post(('/api/collection'), (req, res, next) => {
   const dbQuery = db.query(sql, params);
   dbQuery.then(result => {
     res.status(201).send(result.rows[0]);
-  }).catch(err => {
-    next(err);
-  });
+  }).catch(err => next(err));
+});
+
+app.post(('/api/restaurant'), (req, res, next) => {
+  const { collectionId, businessId } = req.body;
+  if (!collectionId || !businessId) {
+    throw new ClientError(400, 'collectionId and businessId are required');
+  }
+  client.business(businessId)
+    .then(response => {
+      const json = response.jsonBody;
+      const sql = `
+      insert into "restaurants" ("collectionId", "businessId", "json")
+      values ($1, $2, $3)
+      returning *;
+      `;
+      const params = [collectionId, businessId, json];
+      const dbQuery = db.query(sql, params);
+      dbQuery.then(result => {
+        res.status(201).send(result.rows[0]);
+      }).catch(err => next(err));
+      const setImageSQL = `
+        update "collections"
+           set "image" = coalesce("image", $1)
+         where "collectionId" = $2
+      `;
+      const setImageParams = [json.image_url, collectionId];
+      const dbQueryImage = db.query(setImageSQL, setImageParams);
+      dbQueryImage.then()
+        .catch(err => next(err));
```

Comment:
These should be moved into your first `query().then()` callback so that the whole thing completes before you respond.